### PR TITLE
Added headers

### PR DESCRIPTION
**Pull request by truthfool**
## Description

- added needed headers to spring api

 _(Ignore if these details are present on the associated JIRA ticket)_

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [ ] Design related bullet points or design document link related to this PR added in the description above. 

- [ ] Updated corresponding Postman Collection or Api documentation for the changes in this PR.

- [ ] Create/update unit or integration tests for verifying the changes made.

- [ ] Add required Swagger annotation and update API documentation with details of any API changes if applicable

- [ ] Followed the naming conventions as given in https://docs.google.com/document/d/1Q4vaMSzrTxxh9TS0RILuNkSkYCxotuYk1Xe0CMIkkCU/edit?usp=sharing

### CODE REVIEW

**Comment by truthfool** [comment]
@avikganguly01 @fynmanoj

File: src/main/java/org/mifos/connector/channel/api/definition/GSMATransactionApi.java [hunk 1]
```diff
@@ -10,5 +10,7 @@ public interface GSMATransactionApi {
     GsmaP2PResponseDto gsmatransaction(@RequestBody GsmaTransfer requestBody,
                                        @RequestHeader(value = "X-CorrelationID") String correlationId,
                                        @RequestHeader(value = "amsName") String amsName,
-                                       @RequestHeader(value = "accountHoldingInstitutionId") String accountHoldId) throws JsonProcessingException;
+                                       @RequestHeader(value = "accountHoldingInstitutionId") String accountHoldId,
```
Comment (fynmanoj) [inline src/main/java/org/mifos/connector/channel/api/definition/GSMATransactionApi.java:13]: you cannot add these headers here. Remember we discussed that these headers should not appear in the channel connector code.  This code is generic not mpesa specific. we cannot add any header that is solution specific

about  reconciled and mpesaTxnId

File: src/main/java/org/mifos/connector/channel/api/implementation/GSMATransactionApiController.java [hunk 2]
```diff
@@ -25,11 +25,13 @@ public class GSMATransactionApiController implements GSMATransactionApi {
...
 
     @Override
-    public GsmaP2PResponseDto gsmatransaction(GsmaTransfer requestBody, String correlationId,String amsName,String accountHoldId) throws JsonProcessingException {
+    public GsmaP2PResponseDto gsmatransaction(GsmaTransfer requestBody, String correlationId,String amsName,String accountHoldId,String reconciled,String mpesaTxnId) throws JsonProcessingException {
         Headers headers = new Headers.HeaderBuilder()
                 .addHeader("X-CorrelationID", correlationId)
                 .addHeader("amsName",amsName)
                 .addHeader("accountHoldingInstitutionId",accountHoldId)
+                .addHeader("reconciled",reconciled)
+                .addHeader("mpesaTxnId",mpesaTxnId)
```
Comment (fynmanoj) [inline src/main/java/org/mifos/connector/channel/api/implementation/GSMATransactionApiController.java:34]: remove these headers from channel connector

**Review by fynmanoj — CHANGES_REQUESTED** [review]
Solution specific code cannot be added to Generic API , GSMA transacton

File: src/main/java/org/mifos/connector/channel/api/definition/GSMATransactionApi.java (hunk 1 above)
Comment (truthfool) [reply src/main/java/org/mifos/connector/channel/api/definition/GSMATransactionApi.java:13]: This is needed further in mpesa- connector for settlement flow.

File: src/main/java/org/mifos/connector/channel/api/implementation/GSMATransactionApiController.java (hunk 2 above)
Comment (truthfool) [reply src/main/java/org/mifos/connector/channel/api/implementation/GSMATransactionApiController.java:34]: This is needed further in mpesa- connector for settlement flow.

**Pull request merged**